platform/default/mbgl/storage/offline_download.cpp

```diff
@@ -230,6 +230,7 @@ void OfflineDownload::ensureResource(const Resource& resource, std::function<voi
             && util::mapbox::isMapboxURL(resource.url)
             && offlineDatabase.offlineMapboxTileCountLimitExceeded()) {
             observer->mapboxTileCountLimitExceeded(offlineDatabase.getOfflineMapboxTileCountLimit());
+            setState(OfflineRegionDownloadState::Inactive);
```

Comment:
Should this be flipped? Now `mapboxTileCountLimitExceeded()` should be a signal to client code that the download has stopped, not that it will stop. (For context, in the iOS SDK implementation in #4221, I’d like to set `MGLOfflineTask.state` to `MGLOfflineTaskStateInactive` inside the callback.)